code-challenges.test.js

```diff
@@ -32,8 +94,59 @@ const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
 const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
 // Expected output: [ 2, 1, -1 ]
 
+
+describe("OnlyTheRemaindersIfYouPlease", () => {
+  it("returns an array of only the remainders of the numbers when divided by 3", () => {
+    expect(OnlyTheRemaindersIfYouPlease(hodgepodge1)).toEqual([ 2, 0, -1, 0 ])
+    expect(OnlyTheRemaindersIfYouPlease(hodgepodge2)).toEqual([ 2, 1, -1 ])
+    
+  })
+})
+// FAIL  ./code-challenges.test.js
+// OnlyTheRemaindersIfYouPlease
+//   ✕ returns an array of only the remainders of the numbers when divided by 3 (1 ms)
+
+// ● OnlyTheRemaindersIfYouPlease › returns an array of only the remainders of the numbers when divided by 3
+
+//   ReferenceError: OnlyTheRemaindersIfYouPlease is not defined
+
+
+
+
 // b) Create the function that makes the test pass.
 
+// Psuedo: create a function called OnlyTheRemaindersIfYouPlease. What does this function do ? it takes in an array containing mixed data types(input) and it will return an array with only the remainders of the numbers when divided by 3 . How do I accomplish this ? using an built-in method to iterate over the array and return the specific values I'm looking for . I can use the .filter method which filters out any non-numeric elements from the input array using the typeof operator (element === 'number' to specify that I want to return only numbers) and isNaN() (which will check if each element is a valid number or not. This function returns true if the value is not a number, and false if it is a number.) My function then uses the Array.map() method to generate an array of the remainders of those numbers when divided by 3.
+// refactor---> My function takes an array of mixed data as its argument. It then filters out any non-numeric elements from the array using the filter method and checks if each element is a valid number using the typeof operator and isNaN() function. It then applies the modulo operator (%) to each numeric element to get its remainder when divided by 3. Finally, it returns an array containing the remainders of all the valid numeric elements in the original array. 
+// finally refactor--> Basically, the function takes an array, removes non-numeric elements, and returns an array of remainders of the numeric elements in the input array when divided by 3. The filter() method takes a callback function that gets executed on each element of the input array. This callback function checks whether the element is a valid number and is not NaN, using the typeof operator and isNaN() function, respectively. If the element passes both of these checks, it is kept in the resulting array returned by the filter() method.Therefore, any NaN values or non-numeric values will be removed from the input array before calculating the remainders.
+
+
+// //  attempt 3 (final refactor)
+const OnlyTheRemaindersIfYouPlease = (array) => {
```

Comment:
I am happy to see you are using es6 arrow syntax here, but make sure that you are using camelCasing for your function name.  PascalCasing is used primarily when working with classes.